appcelerator-archive/amp
# Cluster deployment on AWS

This PR adds the missing elements for AMP stack deployment on AWS.

- symlink for hack/deploy: hack/deploy-aws. The script is now able to deploy locally or on AWS
- the terraform deployment now has several outputs: the cluster id, the leader manager ip and all the ips of the bootstrapped instances.

to test:

- you should have aws credentials set in your home directory (~/.aws/credentials). Please refer to http://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html#credentials-file-format.
- an aws keypair should be available in this region.
- add a terraform.tfvars in the bootstrap/terraform-aws folder with a content similar to
```
aws_name = "TAG"
aws_region = "us-west-2"
bootstrap_key_name = "AWS-KEYPAIR-ID"
aws_profile = "default"
infrakit_config_base_url = "https://raw.githubusercontent.com/appcelerator/amp/cluster-aws/bootstrap"
```
The key name should exist in AWS EC2 in this region and you should have the private key in your possession.
- run `hack/deploy-aws`
- while the script is running, the cluster id and the manager leader ip are printed
- add the leader IP (:5000) as an insecure registry on your host, restart the engine
- if the script is still running and has not tried to push images yet, keep waiting, else run `./hack/deploy-aws $clusterid`
- it should deploy the stacks on the cluster and validate the availability of nats and elasticsearch
- get the dns name of one of the node, and connect on port 50106, kibana should appear.
